src/local_device.rs

```diff
@@ -88,9 +89,24 @@ impl LocalDevice {
             [user_manifest_id: EntryID, "user_manifest_id"],
             [user_manifest_key: SecretKey, "user_manifest_key"],
             [local_symkey: SecretKey, "local_symkey"],
+            [time_provider: TimeProvider, "time_provider"],
+        );
+        crate::binding_utils::check_mandatory_kwargs!(
+            [organization_addr, "organization_addr"],
+            [device_id, "device_id"],
+            [device_label, "device_label"],
+            [human_handle, "human_handle"],
+            [signing_key, "signing_key"],
+            [private_key, "private_key"],
+            [profile, "profile"],
+            [user_manifest_id, "user_manifest_id"],
+            [user_manifest_key, "user_manifest_key"],
+            [local_symkey, "local_symkey"],
+            // time_provider is not mandatory
         );
```

Comment:
Why you don't `parse_kwargs_optional!` for `time_provider` instead of `create check_mandatory_kwargs!` macro ?

Reply:
`parse_kwargs_optional!` raises an error if it encounter an unknown param
So I would have to provide all the params to `parse_kwargs_optional!` (an not just the single one which is optional)

Reply:
but `parse_kwargs` do the same too